.circleci/config.yml

```
@@ -0,0 +1,314 @@
+        CIRCLE_ARTIFACTS: /Users/distiller/code/cordova/platforms/ios/build/device
+
+    steps:
+      - checkout
+      - run: *yarn
+      - save-cache: *save-yarn-cache
+      - run: yarn run bower install
+      - save-cache: *save-bower-cache
+      - run:
+          name: Setup Environment variables
+          command: |
+            echo 'export PATH=$PATH:`yarn bin`' >> $BASH_ENV
+            echo 'export ENVIRONMENT=$(if [ "$CIRCLE_BRANCH" == "live" ]; then echo production; else echo staging; fi)' >> $BASH_ENV
+            echo 'export PROVISIONING_PROFILE=$(if [ "$CIRCLE_BRANCH" == "live" ]; then echo $ENCODED_PROD_PROVISIONING_PROFILE; else echo $ENCODED_STAGING_PROVISIONING_PROFILE; fi)' >> $BASH_ENV
+            echo 'export CERTIFICATES=$(if [ "$CIRCLE_BRANCH" == "live" ]; then echo $ENCODED_PROD_CERT; else echo $ENCODED_STAGING_CERT; fi)' >> $BASH_ENV
+      - run:
+          command: |
+              npm rebuild node-sass
+              yarn run ember build --environment=production
+      - run: mv dist/ cordova/www
+      - persist_to_workspace:
+          root: .
+          paths:
+            - cordova/www
+      - run: bundle check || bundle install --path vendor/bundle
+      - save-cache: *save-bundle-cache
+      - attach_workspace:
+          at: ~/code
+      - run:
+          name: make Provisioning Profiles directory
+          command: mkdir -pv ~/Library/MobileDevice/Provisioning\ Profiles/
+      - run:
+          name: decode Provisioning Profiles
+          context: org-global
+          command: base64 -D -o ~/Library/MobileDevice/Provisioning\ Profiles/App_Store.mobileprovision <<< "${PROVISIONING_PROFILE}"
+      - run:
+          name: install cordova-update-config globally
+          command: npm i -g cordova-update-config
+      - run:
+          name: install cordova globally
+          command: npm i -g cordova@6.5.0
+      - run:
+          name: create keychain and import certificates
+          command: bundle exec fastlane ios prepare_certificates cert:$CERTIFICATES
+          working_directory: ~/code
+      - run:
+          name: iOS build
+          command: bundle exec rake ${ENVIRONMENT} ios app:build
+          working_directory: cordova
+      - store_artifacts:
+          path: cordova/platforms/ios/build/device
+      - run:
+          name: release ios build
+          command: bundle exec fastlane ios ${ENVIRONMENT}
+          working_directory: ~/code
+
+
+workflows:
+    version: 2
+    build_and_bundle:
+      jobs:
+        - checkout_code
+        - package_dependencies:
+            requires:
+              - checkout_code
+        - test_cases:
+            requires:
+              - package_dependencies
+        - upload-coverage:
+            requires:
+              - package_dependencies
+              - test_cases
+            filters:
+              branches:
+                only: /^(master|live)$/
+        - www_build:
+            requires:
+              - upload-coverage
```

I would switch this to require `test_cases` as upload_coverage isn't a hard dependency.

```suggestion
              - test_cases
```